<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>582</width>
    <height>683</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ChatRoom</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>561</width>
      <height>641</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="page">
     <widget class="QLabel" name="genderlabel">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>300</y>
        <width>321</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>25</pointsize>
       </font>
      </property>
      <property name="text">
       <string>請選擇性別開始聊天</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="connectButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>560</y>
        <width>120</width>
        <height>60</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>開始聊天</string>
      </property>
     </widget>
     <widget class="QPushButton" name="girlButton">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>40</y>
        <width>120</width>
        <height>120</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="source.qrc">
        <normaloff>:/newPrefix/girl_icon.png</normaloff>:/newPrefix/girl_icon.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>120</width>
        <height>120</height>
       </size>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="boyButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>40</y>
        <width>120</width>
        <height>120</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="source.qrc">
        <normaloff>:/newPrefix/boy_icon.png</normaloff>:/newPrefix/boy_icon.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>120</width>
        <height>120</height>
       </size>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="over18Mode">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>370</y>
        <width>301</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="text">
       <string>18+ Mode</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <widget class="QPushButton" name="backButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>30</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;-</string>
      </property>
     </widget>
     <widget class="QWidget" name="chatWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>531</width>
        <height>591</height>
       </rect>
      </property>
      <widget class="QLabel" name="genderPicClient">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>170</y>
         <width>61</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="source.qrc">:/newPrefix/boy.png</pixmap>
       </property>
      </widget>
      <widget class="QLabel" name="genderPicUser">
       <property name="geometry">
        <rect>
         <x>430</x>
         <y>390</y>
         <width>61</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="source.qrc">:/newPrefix/boy.png</pixmap>
       </property>
      </widget>
      <widget class="QLabel" name="chatBubble">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>290</y>
         <width>421</width>
         <height>181</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="source.qrc">:/newPrefix/talking.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="chatBubble_2">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>-60</y>
         <width>431</width>
         <height>261</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="source.qrc">:/newPrefix/talking_2.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QTextBrowser" name="userText">
       <property name="geometry">
        <rect>
         <x>105</x>
         <y>330</y>
         <width>241</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="styleSheet">
        <string notr="true">border-color: rgba(0, 0, 0, 0);
background-color: rgba(255, 255, 255, 0);
border-width : 1.2px;
border-style:inset;
</string>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'PMingLiU'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QPushButton" name="sendButton">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>510</y>
         <width>80</width>
         <height>60</height>
        </rect>
       </property>
       <property name="text">
        <string>傳送</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="textArea">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>510</y>
         <width>401</width>
         <height>60</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'PMingLiU'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="userName_2">
       <property name="geometry">
        <rect>
         <x>420</x>
         <y>360</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>你</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="clientName">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>140</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>對方</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="clientText">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>50</y>
         <width>251</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="waitingWidget" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>551</width>
        <height>601</height>
       </rect>
      </property>
      <widget class="QLabel" name="waitingText">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>190</y>
         <width>441</width>
         <height>141</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string>等待其他人加入中....</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="reconnectButton">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>460</y>
         <width>120</width>
         <height>60</height>
        </rect>
       </property>
       <property name="text">
        <string>重新連線</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="errorMessage">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>475</y>
      <width>441</width>
      <height>121</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>無法連接伺服器請稍後再試一次</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>582</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="source.qrc"/>
  <include location="source.qrc"/>
  <include location="source.qrc"/>
  <include location="../../../.designer/backup/source.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>backButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>exitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>29</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>102</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>492</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>boyButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>boyClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>134</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>girlButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>girlClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>410</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connectButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>connectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>520</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reconnectButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>connectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>557</y>
    </hint>
    <hint type="destinationlabel">
     <x>573</x>
     <y>608</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>boyClicked()</slot>
  <slot>girlClicked()</slot>
  <slot>connectClicked()</slot>
  <slot>exitClicked()</slot>
  <slot>sendClicked()</slot>
  <slot>nameEditFinish()</slot>
 </slots>
</ui>
